app/services/authentication/authenticator.rb

```diff
@@ -151,7 +151,16 @@ def default_user_fields
     def update_user(user)
       user.tap do |model|
         model.unlock_access! if model.access_locked?
-        model.assign_attributes(provider.existing_user_data)
+
+        if model.confirmed?
+          # We don't want to update users' email or any other fields if they're
+          # connecting an existing account that already has a confirmed email.
+          model.assign_attributes(provider.existing_user_data.except(:email))
+        else
+          # If the user doesn't have a confirmed email we can update their email
+          # and trust it because the auth provider confirmed email ownership
+          model.assign_attributes(provider.existing_user_data)
+        end
```

Comment:
To provide a little more clarity to reviewers that haven't visited this part of the codebase recently (I hadn't been here recently enough to remember the payload structures 😅 ):

This is what printing `model`, `provider` and then `provider.existing_user_data` look like right before these lines

<img width="1440" alt="Screen Shot 2021-11-30 at 15 35 29" src="https://user-images.githubusercontent.com/6045239/144143509-7ca91bc6-d1dd-42d3-94e8-19af96d28f70.png">

So the logic here is to drop the email (`.except`) in case the user is already `confirmed?`, because we do want to assign the other attributed (i.e. `<provider_name>_username`).